File: templates/javascript/v1/designtemplates/iterator/collection.js
export class ItemCollection {
  constructor() {
    this.items = [];
  }

  add(item) {
    this.items.push(item);
  }

  [Symbol.iterator]() {
    let index = 0;
    const items = this.items;
    return {
      next() {
        if (index < items.length) {
          return { value: items[index++], done: false };
        }
        return { value: undefined, done: true };
      }
    };
  }
}

File: templates/python/v1/designtemplates/iterator/collection.py
from __future__ import annotations

from dataclasses import dataclass, field
from typing import Iterator, List


@dataclass(slots=True)
class Item:
    id: str
    name: str


@dataclass(slots=True)
class ItemCollection:
    items: List[Item] = field(default_factory=list)

    def add(self, item: Item) -> None:
        self.items.append(item)

    def __iter__(self) -> Iterator[Item]:
        return iter(self.items)
